app/src/main/java/com/example/microfeaturecodelab/presentation/MainActivity.kt

```diff
@@ -36,40 +56,112 @@ class MainActivity : ComponentActivity() {
+}
+
 @Composable
 fun JobScreen(
     modifier: Modifier = Modifier,
     viewModel: JobsViewModel = hiltViewModel()
 ) {
-    viewModel.hashCode()
-    Column(modifier.fillMaxSize()) {
-        Text("Job Screen 1", Modifier.align(Alignment.CenterHorizontally))
-        Text("Job Widget 2", Modifier.align(Alignment.CenterHorizontally))
-        JobRecommendation(viewModel.feature, Modifier.align(Alignment.CenterHorizontally))
-        // Filter
-        // Job picks for you
-        // Enable Job updates
-        // Recent Searches
-        // Explore company carousel
-        // Job list type1
-        // Discover linkedin
-        // More job paginated list
+    val components by remember { mutableStateOf(viewModel.getComponents()) }
+    StableWidget(components, viewModel.componentViewModels, modifier = modifier)
+}
+
+
+@Composable
+fun StableWidget(
+    items: Widgets,
+    map: Map<String, MicroFeatureViewModel>,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = PaddingValues(16.dp),
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        // Dynamically generate items for components
+        items(
+            items = items.items,
+            key = { it.id },
+            itemContent = {
+                val componentViewModel = remember(items) {
+                    map[it.id]
+                }
+                Log.d("MicroFeature", "JobScreen:index ${it.id}")
+                ComponentItem(it, componentViewModel)
+            }
+        )
+    }
+}
+
+@Composable
+fun ComponentItem(
+    component: ComponentConfig,
+    componentViewModel: MicroFeatureViewModel?
+) {
+    when (component.type) {
+        "personalisedJob" -> {
+
+            JobRecommendation(
+                componentViewModel as PersonalisedJobViewModel,
+                component.id,
+                Modifier.fillMaxWidth()
+            )
+        }
     }
 }
 
 @Composable
 fun JobRecommendation(
     viewmodel: PersonalisedJobViewModel,
+    id: String,
     modifier: Modifier = Modifier
 ) {
-    val personalisedJob by viewmodel.uiState.collectAsStateWithLifecycle()
-    when (personalisedJob) {
+    Log.d("MicroFeature", "JobRecommendation: Recomposing Before collect item Id: $id")
+
+    // This collect triggers the flow in the viewmodel to fetch data from use case
+    // @TODO Currently its being called for all 20 items in lazy column which is inefficient
+
+    // Uncomment this to see the issue, with sealed state
+    // val personalisedJob by viewmodel.uiState.collectAsStateWithLifecycle()
+    // StateReducer(id, personalisedJob, modifier = modifier)
+
+    // Working fine when uistate is directly consumed
+    val personalisedJob by viewmodel.uiStateWorking.collectAsStateWithLifecycle()
+    Log.d("MicroFeature", "StateReducer: recomposing $id")
+    JobRecommendationSection(personalisedJob, modifier = modifier)
```

Comment:
Issue with lazy column not doing lazy recompositions when consuming a sealed state vs consuming the state directly